### ISSUE TITLE
Exclude the X-Pack Security plugin from the deprecation warning for REST wrappers 

### ISSUE THREAD

**Issue by bytebilly** [elastic/elasticsearch#73227]
In https://github.com/elastic/elasticsearch/pull/70523 a deprecation warning has been introduced for any plugin that installs a REST wrapper.

The code includes a check to avoid raising the warning for Elasticsearch official plugins, like the X-Pack Security plugin, but the deprecation warning is raised anyway.

```
{
  "type": "deprecation.elasticsearch",
  "timestamp": "2021-05-19T08:23:01,649Z",
  "level": "DEPRECATION",
  "component": "o.e.d.a.ActionModule",
  "cluster.name": "elasticsearch",
  "node.name": "fabio-enterprise-test",
  "message": "The org.elasticsearch.xpack.security.Securityplugin installs a custom REST wrapper. This functionality is deprecated and will not be possible in Elasticsearch 8.0. If this plugin is intended to provide security features for Elasticsearch then you should switch to using the built-in Elasticsearch features instead."
}
```

Cluster info: 

```
{
  "name" : "fabio-enterprise-test",
  "cluster_name" : "elasticsearch",
  "cluster_uuid" : "gri5DCocRXSP_sMT-59EgQ",
  "version" : {
    "number" : "7.13.0-SNAPSHOT",
    "build_flavor" : "default",
    "build_type" : "tar",
    "build_hash" : "7a060b2776ca4a6fb7127ffac8a4b6ebf2cd8cda",
    "build_date" : "2021-05-19T01:30:24.861016826Z",
    "build_snapshot" : true,
    "lucene_version" : "8.8.2",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "You Know, for Search"
}
```
cc @BigPandaToo

**Comment by colings86** [comment]
I just ran `docker run -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:7.13.0-SNAPSHOT` locally and got what appears to be the same build commit but could not reproduce the issue. Response from "/":
```
{
name: "f7b2eb618f8b",
cluster_name: "docker-cluster",
cluster_uuid: "gM1-3peKRkuql3pfGhw59g",
version: {
number: "7.13.0-SNAPSHOT",
build_flavor: "default",
build_type: "docker",
build_hash: "7a060b2776ca4a6fb7127ffac8a4b6ebf2cd8cda",
build_date: "2021-05-19T01:30:24.861016826Z",
build_snapshot: true,
lucene_version: "8.8.2",
minimum_wire_compatibility_version: "6.8.0",
minimum_index_compatibility_version: "6.0.0-beta1"
},
tagline: "You Know, for Search"
}
```

**Comment by BigPandaToo** [comment]
Resolved by  #73249

**Comment by BigPandaToo** [comment]
Resolved by #73249
